app/services/services.py

```diff
@@ -299,10 +324,21 @@ async def generate_transcript(audio_file: str, save_to: str, api_key: str):
         )
         deepgram.extra.to_SRT(response)
 
-        convert_to_srt(response, save_to)
+        transcript_file = f"{save_to}.{file_format}"
+
+        if file_format == "srt":
+            transcript_file = convert_to_srt(response, transcript_file)
+        elif file_format == "json":
+            transcript_file = convert_to_json(response, transcript_file)
+        else:
+            raise HTTPException(
+                status_code=400, detail="Unsupported file format"
+            )
+
+    return transcript_file
```

Comment:
```suggestion
        if file_format not in ("srt", "json"):
            raise HTTPException(
                status_code=400, detail="Unsupported file format"
            )
            
        if file_format == "srt":
            return convert_to_srt(response, transcript_file)
        elif file_format == "json":
            return convert_to_json(response, transcript_file)
```
There's also another approach to reduce the even remove the "elif".
We'll fix it later (hopefully)

Reply:
Thanks, great suggestion